From 5a7b11a1883ca49b8bea37c56524864b4f47449f Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <josh@wilsdon.ca>
Date: Thu, 20 Dec 2018 13:13:05 -0800
Subject: [PATCH] TRITON-1055 sdcadm experimental update-agents with
 --just-download flag fails on SHA1 Reviewed by: Todd Whiteman
 <todd.whiteman@joyent.com> Approved by: Todd Whiteman
 <todd.whiteman@joyent.com>

---
 CHANGES.md                  | 4 ++++
 lib/cli/do_update_agents.js | 5 +++++
 package.json                | 2 +-
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/CHANGES.md b/CHANGES.md
index cc2a6c4..3e88163 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,10 @@
 
 # sdcadm Changelog
 
+## 1.24.2
+
+- TRITON-1055 sdcadm experimental update-agents with --just-download flag fails on SHA1
+
 ## 1.24.1
 
 - TOOLS-2117 sdcnode updates for November 2018 node.js security releases
diff --git a/lib/cli/do_update_agents.js b/lib/cli/do_update_agents.js
index 226c7b5..45315b0 100644
--- a/lib/cli/do_update_agents.js
+++ b/lib/cli/do_update_agents.js
@@ -707,6 +707,11 @@ UpdateAgents.prototype.exec = function exec(callback) {
                 ctx.absfname
             ];
 
+            if (self.justDownload) {
+                next();
+                return;
+            }
+
             common.execFilePlus({
                 argv: argv,
                 log: self.sdcadm.log
diff --git a/package.json b/package.json
index c5745f7..b5b4ea0 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.24.1",
+  "version": "1.24.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

